### PR TITLE
Added support Helix 2.0

### PR DESCRIPTION
**Pull request by TanaTau**

### CODE REVIEW

File: generators/app/index.js [hunk 1]
```diff
@@ -80,6 +81,12 @@ module.exports = class HelixGenerator extends BaseGenerator {
 
     this.options.solutionNameUri = this.options.solutionName.replace(/[^a-z0-9\-]/ig, '-').toLowerCase();
     this.config.set('solutionNameUri', this.options.solutionNameUri);
+
+    // setup name of code folder
+    var scVersion = this.options.sitecoreUpdate.majorVersion && Number(this.options.sitecoreUpdate.majorVersion) >= 9.3;
```
Comment (ashishagin) [inline generators/app/index.js:86]: Please align variable name with type

File: generators/app/index.js [hunk 2]
```diff
@@ -80,6 +81,12 @@ module.exports = class HelixGenerator extends BaseGenerator {
...
+    this.options.supportHelix20 = scVersion ? true : false;
```
Comment (ashishagin) [inline generators/app/index.js:87]: !!scVersion

File: generators/app/index.js [hunk 3]
```diff
@@ -129,7 +140,9 @@ module.exports = class HelixGenerator extends BaseGenerator {
 
   _processYmlFile(content, path) {
     let result = content instanceof Buffer ? content.toString('utf8') : content;
-    result = result.replace(/SolutionX/g, this.options.solutionName);
+    result = result.replace(/SolutionX/g, this.options.solutionName)
+      .replace(/CodeFolderX/g, this.options.codeFolderName)
+      .replace(/SupportHelix20X/g, this.options.supportHelix20);
```
Comment (ashishagin) [inline generators/app/index.js:145]: Please remove SupportHelix

File: generators/app/index.js [hunk 4]
```diff
@@ -141,7 +154,9 @@ module.exports = class HelixGenerator extends BaseGenerator {
   }
 
   _processPathSolutionToken(destPath) {
-    return destPath.replace(/SolutionX/g, '<%= solutionX %>');
+    return destPath.replace(/SolutionX/g, '<%= solutionX %>')
+      .replace(/CodeFolderX/g, '<%= codeFolderX %>')
+      .replace(/SupportHelix20X/g, '<%= supportHelix20X %>');
```
Comment (ashishagin) [inline generators/app/index.js:159]: Please remove SupportHelix

File: generators/app/templates/default/src/build.cake [hunk 5]
```diff
@@ -17,6 +17,7 @@ Sitecore.Constants.SetNames();
 Sitecore.Parameters.InitParams(
     context: Context,
     msBuildToolVersion: MSBuildToolVersion.Default,
+    supportHelix20: "<%= supportHelix20X %>",
```
Comment (ashishagin) [inline generators/app/templates/default/src/build.cake:20]: Please unwrap from quotes

File: generators/module/index.js [hunk 6]
```diff
@@ -110,6 +111,14 @@ module.exports = class extends BaseGenerator {
     } else if (this.options.moduleType == 'Feature') {
       this.options.unicornSerializationDependenciesX = this.options.solutionName + '.Foundation.*';
     }
+
+    // setup name of code folder
+    if(this.options.supportHelix20) {
```
Comment (ashishagin) [inline generators/module/index.js:116]: Please revisit

File: generators/app/index.js (hunk 1 above)
Comment (TanaTau) [reply generators/app/index.js:86]: Renamed to isRequiredScVersion

File: generators/app/index.js (hunk 2 above)
Comment (TanaTau) [reply generators/app/index.js:87]: Fixed

File: generators/app/index.js (hunk 3 above)
Comment (TanaTau) [reply generators/app/index.js:145]: Removed

File: generators/app/index.js (hunk 4 above)
Comment (TanaTau) [reply generators/app/index.js:159]: Removed

File: generators/module/index.js (hunk 6 above)
Comment (TanaTau) [reply generators/module/index.js:116]: Modified

File: generators/app/templates/default/src/build.cake (hunk 5 above)
Comment (TanaTau) [reply generators/app/templates/default/src/build.cake:20]: This parameter will be converted from string to boolean later

**Pull request merged**